### PR TITLE
[ci] further simplify CI configurations

### PR DESCRIPTION
**Pull request by jameslamb**
Follow-up to #6458

Proposes the following simplifications to CI configurations and scripts:

* removes `CONDA_ENV` environment variable and just hard-codes it in scripts
* removes configurations to run CI on PRs targeting any branch matching the pattern `release/*`
   - *this was added back when we were releasing from branches not targeting `master`, to avoid releasing the breaking changes building up there in anticipation of 4.0... but that's not the state this project is in any more*
* consolidates `conda`-setup code in `test.sh`
* reduces duplication of image URI in CUDA CI job config
* removes a Python 2.7 try-catch import in `.ci/get_workflow_status.py`
* stops writing an env file in QEMU_multiarch job, in favor of just using `docker run --env`

Also increases the valgrind allow-possibly-lost-bytes limit (see https://github.com/microsoft/LightGBM/pull/6213#issuecomment-2131463267)

### CODE REVIEW

File: .github/workflows/cuda.yml
Comment (borchero) [inline .github/workflows/cuda.yml:65]:
```suggestion
      image: nvcr.io/nvidia/cuda:${{ matrix.cuda_version }}-devel-${{ matrix.linux_version }}
```

File: .github/workflows/cuda.yml
Comment (borchero) [inline .github/workflows/cuda.yml:86]: Remove the `image` here?

**Review by borchero — APPROVED** [review]
LGTM!

File: .github/workflows/cuda.yml
Comment (jameslamb) [reply .github/workflows/cuda.yml]: ah yep, missed this one. Thank you!

Removed in https://github.com/microsoft/LightGBM/pull/6463/commits/e8b679ab880f802feede466d48da7702f84a751c

**Comment by jameslamb** [comment]
I pushed a `pydistcheck` release last night that broke CI here.

```text
ERROR: found the following unrecognized checks passed via '--ignore': compiled-objects-have-debug-symbols,distro-too-large-compressed
```

([build link](https://github.com/microsoft/LightGBM/actions/runs/9270347304/job/25503157540))

Just pushed a commit fixing that: https://github.com/microsoft/LightGBM/pull/6463/commits/f99a01eda493323f4041aaf79690ead3d195c069

It's so minor and noncontroversial, I'm going to merge this if CI passes, without waiting for another review. We can revert it easily if anyone comes back and disagrees with the changes I made.

**Pull request merged**
